nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -158,6 +158,7 @@ let
           server_name ${serverName} ${concatStringsSep " " vhost.serverAliases};
           ${acmeLocation}
           ${optionalString (vhost.root != null) "root ${vhost.root};"}
+          ${optionalString (vhost.alias != null) "alias ${vhost.alias};"}
```

Comment:
Have you checked whether alias works in a `server`? According to the [docs](https://nginx.org/en/docs/http/ngx_http_core_module.html#alias), it should only work in `location` blocks.